### ISSUE TITLE
[riverpod_generator] 2.3.4 getter issues with `createdType` and `valueType`

### ISSUE THREAD

**Issue by dantereve** [rrousselGit/riverpod#3024]
Hello,

I've faced a little issue with `riverpod_generator` 2.3.4.

I've installed riverpod's related packages on one of the feature package of my app using this command: 

```shell
flutter pub add hooks_riverpod
flutter pub add flutter_hooks
flutter pub add riverpod_annotation
flutter pub add dev:riverpod_generator
flutter pub add dev:build_runner
flutter pub add dev:custom_lint
flutter pub add dev:riverpod_lint
```

Which installed `riverpod_generator` 2.3.4.

```dart
dependencies:
  flutter_hooks: ^0.20.3
  hooks_riverpod: ^2.4.4
  riverpod_annotation: ^2.2.1

dev_dependencies:
  build_runner: ^2.4.6
  custom_lint: ^0.5.4
  riverpod_generator: ^2.3.4
  riverpod_lint: ^2.3.1
```

Then I tried to run the build runner for this provider in watch mode:

```dart
import 'package:riverpod_annotation/riverpod_annotation.dart';

part 'current_index_controller.g.dart';

@riverpod
class CurrentIndexController extends _$CurrentIndexController {
  @override
  int build({int initialIndex = 0}) {
    return initialIndex;
  }
}
```
The build runner command gave me this result: 

```
flutter pub run build_runner watch --delete-conflicting-outputs
[INFO] Generating build script... 
[INFO] Generating build script completed, took 193ms  
[INFO] Precompiling build script...... 
[WARNING] ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/functional_provider.dart:38:33: Error: The getter 'createdType' isn't defined for the class 'FunctionalProviderDeclaration'.  - 'FunctionalProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'createdType'.     final returnType = provider.createdType;                                 ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/functional_provider.dart:61:48: Error: The getter 'valueType' isn't defined for the class 'FunctionalProviderDeclaration'.  - 'FunctionalProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'. final $providerName = $providerType<${provider.valueType}>.internal(                                                ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/functional_provider.dart:69:50: Error: The getter 'valueType' isn't defined for the class 'FunctionalProviderDeclaration'.  - 'FunctionalProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'. typedef $refName = ${providerType}Ref<${provider.valueType}>;                                                  ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/class_based_provider.dart:101:33: Error: The getter 'createdType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'createdType'.     final returnType = provider.createdType;                                 ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/class_based_provider.dart:115:66: Error: The getter 'valueType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'. final $providerName = $providerType<${provider.name}, ${provider.valueType}>.internal(                                                                  ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/class_based_provider.dart:123:61: Error: The getter 'valueType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'. typedef $notifierTypedefName = $notifierBaseType<${provider.valueType}>;                                                             ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:55:32: Error: The getter 'createdType' isn't defined for the class 'FunctionalProviderDeclaration'.  - 'FunctionalProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'createdType'.     var createdType = provider.createdType.toString();                                ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:57:33: Error: The getter 'createdType' isn't defined for the class 'FunctionalProviderDeclaration'.  - 'FunctionalProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'createdType'.     final returnType = provider.createdType;                                 ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:65:44: Error: The getter 'valueType' isn't defined for the class 'FunctionalProviderDeclaration'.  - 'FunctionalProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'.         createdType = 'FutureOr<${provider.valueType}>';                                            ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:90:38: Error: The getter 'valueType' isn't defined for the class 'FunctionalProviderDeclaration'.  - 'FunctionalProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'.       providerGenerics: '<${provider.valueType}>',                                      ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:134:33: Error: The getter 'createdType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'createdType'.     final returnType = provider.createdType;                                 ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:167:56: Error: The getter 'valueType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'.       providerGenerics: '<${provider.name}, ${provider.valueType}>',                                                        ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:172:74: Error: The getter 'valueType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/river
pod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'. abstract class $notifierTypedefName extends $notifierBaseType<${provider.valueType}> {                                                                          ^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:175:14: Error: The getter 'createdType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'createdType'.   ${provider.createdType} build($parameterDefinition);              ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:180:14: Error: The getter 'createdType' isn't defined for the class 'ClassBasedProviderDeclaration'.  - 'ClassBasedProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'createdType'.   ${provider.createdType} runNotifierBuild(              ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:248:45: Error: The getter 'exposedType' isn't defined for the class 'GeneratorProviderDeclaration'.  - 'GeneratorProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'exposedType'. class $familyName extends Family<${provider.exposedType}> {                                             ^^^^^^^^^^^ ../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_generator-2.3.4/lib/src/templates/family.dart:332:43: Error: The getter 'valueType' isn't defined for the class 'GeneratorProviderDeclaration'.  - 'GeneratorProviderDeclaration' is from 'package:riverpod_analyzer_utils/src/riverpod_ast.dart' ('../../../../../../../../../.pub-cache/hosted/pub.dev/riverpod_analyzer_utils-0.4.1/lib/src/riverpod_ast.dart'). Try correcting the name to the name of an existing getter, or defining a getter or field named 'valueType'. mixin $refNameImpl on $refType<${provider.valueType}> {                                           ^^^^^^^^^ 
[INFO] Precompiling build script... completed, took 524ms  
[SEVERE] Failed to precompile build script .dart_tool/build/entrypoint/build.dart. This is likely caused by a misconfigured builder definition.  
```

This page on pub.dev : https://pub.dev/packages/riverpod_generator/score shows the same getter issue in the static analysis part.

If I downgrade this riverpod_generator to 2.3.3 and riverpod_lint to 2.1.1, everything runs fine again.

```
  riverpod_generator: 2.3.3
  riverpod_lint: 2.1.1
```

```
flutter pub run build_runner watch --delete-conflicting-outputs
[INFO] Generating build script... 
[INFO] Generating build script completed, took 214ms  
[INFO] Precompiling build script...... 
[INFO] Precompiling build script... completed, took 4.2s  
[INFO] Setting up file watchers... 
[INFO] Setting up file watchers completed, took 5ms  
[INFO] Waiting for all file watchers to be ready... 
[INFO] Waiting for all file watchers to be ready completed, took 200ms  
[INFO] Initializing inputs 
[INFO] Building new asset graph... 
[INFO] Building new asset graph completed, took 746ms  [INFO] Checking for unexpected pre-existing outputs.... 
[INFO] Checking for unexpected pre-existing outputs. completed, took 0ms  
[INFO] Running build... 
[INFO] Generating SDK summary... 
[INFO] 1.1s elapsed, 0/8 actions completed. 
[INFO] 2.3s elapsed, 0/8 actions completed. 
[INFO] Generating SDK summary completed, took 2.5s  
[INFO] Running build completed, took 2.6s  [INFO] Caching finalized dependency graph... 
[INFO] Caching finalized dependency graph completed, took 35ms  
[INFO] Succeeded after 2.6s with 0 outputs (16 actions)  
```

If I can be of any help, please, let me know 💪.

**Comment by RipplesCode** [comment]
name: test
description: A new Flutter project.
publish_to: "none" 
version: 1.0.0+1

environment:
  sdk: ">=3.1.4 <4.0.0"

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.2
  flutter_riverpod: ^2.4.4
  riverpod_annotation: ^2.2.1

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^3.0.0
  riverpod_generator: ^2.3.3
  build_runner: ^2.4.6

flutter:

  uses-material-design: true

Getting the below error with above dependencies in pubspec.yaml
[INFO] Precompiling build script... completed, took 7.1s
[SEVERE] Failed to precompile build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.

**Comment by dt-wonder** [comment]
same issue

**Comment by rrousselGit** [comment]
Fixed with 2.3.5

Make sure to run `pub upgrade`
